File: BasicsOfJava/Encapsulation.java
package BasicsOfJava;

class Atm{
    private int paisa;

    public Atm(int p){
        this.paisa=paisa + p;
    }

    public int gatter(int a){
        if(paisa>0 && paisa<a){
        paisa=paisa-a;
       
    }
    return paisa;
}
}

public class Encapsulation {
    public static void main(String[] args) {
        Atm t1=new Atm(100);
        int x=t1.gatter(20);
        System.out.println(x);
        
    }
}

File: BasicOfC++/encapsulation.cpp
#include <iostream>
using namespace std;

class Atm {
private:
    int paisa;   // data hidden (encapsulation)

public:
    // constructor
    Atm(int p) {
        paisa = p;
    }

    // function to withdraw money
    int withdraw(int a) {
        if (paisa >= a) {
            paisa = paisa - a;
        } else {
            cout << "Insufficient Balance" << endl;
        }
        return paisa;
    }
};

int main() {
    Atm t1(100);      // object create
    int x = t1.withdraw(20);  // withdraw
    cout << x;        // output

    return 0;
}